pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -258,7 +258,7 @@ func (exp *flowExporter) sendFlowRecords() error {
 		// (activeFlowTimeout or idleFlowTimeout) are met. A flow is considered
 		// to be idle if its packet counts haven't changed since the last export.
 		if time.Since(record.LastExportTime) >= exp.idleFlowTimeout {
-			if ((record.Conn.OriginalPackets <= record.PrevPackets) && (record.Conn.ReversePackets <= record.PrevReversePackets)) || flowexporter.IsConnectionDying(&record.Conn) {
+			if ((record.Conn.OriginalPackets <= record.PrevPackets) && (record.Conn.ReversePackets <= record.PrevReversePackets)) || flowexporter.IsConnectionDying(&record.Conn) || !record.Conn.IsPresent {
```

Comment:
could we move the check `!record.Conn.IsPresent` into `flowexporter.IsConnectionDying` as a connection not present in the table can be considered as dying too?
Just curious in which scenario is `!record.Conn.IsPresent` required, because we should hit `TIME_WAIT` state `dying status` before the connection gets removed, right?

Reply:
Sure.
I was thinking about the case when `idleTimeOut` or `activeTimeOut` is larger than time for `TIME_WAIT` state so the dying status is not captured. That should happen rarely.

Reply:
Yes it is unlikely. It will happen only when `idleTimeout` is more `TIME_WAIT` state timeout, which by default is 2 mins.